Dev 19

SleeplessShu opened the pull request:
Привет тебе и хорошего дня. Большое спасибо за проверку моей работы.
Пожалуйста, если что-то надо изменить, то подскажи направление в котором мне стоит вносить изменения. Спасибо

SleeplessShu committed:
Спринт 19 коммит 00

- пустой коммит

SleeplessShu committed:
Спринт 19 коммит 01

- принципиальных изменений нет

SleeplessShu committed:
Спринт 19 коммит 02

- все активити кроме плеера заменены на фрагменты

SleeplessShu committed:
Спринт 19 коммит 03

- удалены дублирующиеся и добавлены новые зависимости в gradle

SleeplessShu committed:
Спринт 19 коммит 04

- добавлено нижнее меню
- переработано переключение светлая/тёмная тема

IliaAnK commented on app/src/main/res/layout/mediateka_fragment.xml:
```diff
@@ -5,37 +5,18 @@
-    <LinearLayout
-        android:id="@+id/b_back_to_main"
+
+    <TextView
```
Рекомендация:
В качестве заголовка экрана лучше использовать `Toolbar`. Это стандартный компонент. В него можно передать текст заголовка, иконку(стрелочка влево) с помощью атрибута `app:navigationIcon` и обработать нажатие с помощью `setNavigationOnClickListener {}`. Характеристики текста можно задать через стиль.

Для остальных экранов, аналогично

SleeplessShu replied:
Спасибо. Этот совет вскрыл ещё одну проблему. Я постараюсь реализовать новую вёрстку с новыми знаниями в 20 спринте, тк буду переводить всё на constraint

IliaAnK replied:
@SleeplessShu Окей. 
Единственное отмечу, что не обязательно всё переводить на Сonstraint. К примеру, на мой взгляд, экран "Настройки" лучше оставить на `LinearLayout`.

IliaAnK commented on app/src/main/res/layout/menu_fragment.xml:
```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Рекомендация:
Этого фрагмента больше нет, вёрстку можно тоже удалить

SleeplessShu replied:
сделал

IliaAnK commented:
Рекомендация для вёрстки экрана "Настройки": 
Рекомендация к каждой конструкции
```
<FrameLayout> 
    <TextView/>
    <Switch или ImageView/> 
</FrameLayout>
```
Во-первых, это излишняя вложенность, которая негативно сказывается при отрисовке фрейма.
Во-вторых, она попросту излишня и может вызвать путаницу;
В-третьих, проще формировать и применять стили.

Каждая такая конструкция может быть представлена одной View.
Для свичера:
```
<com.google.android.material.switchmaterial.SwitchMaterial
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:gravity="center_vertical"
    android:paddingHorizontal="16dp"
    android:paddingVertical="20dp"
    android:text="@string/switchDarkTheme"
    android:textSize="16sp"
    android:thumbTintMode="screen" />
```
Для TextView с иконкой справа:
```
<com.google.android.material.textview.MaterialTextView
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:drawableEnd="@drawable/ic_share"
    android:paddingHorizontal="16dp"
    android:paddingVertical="20dp"
    android:text="@string/shareApp"
    android:textSize="16sp" />
```

Общие характеристика вынести само собой можно вынести в стили.

Итого иерархия всего `layout`-а будет вот такой:
```
<LinearLayout>
    <MaterialToolbar/>
    <SwitchMaterial/>
    <MaterialTextView/>
    <MaterialTextView/>
    <MaterialTextView/>
<LinearLayout/>
```

SleeplessShu committed:
Спринт 19 коммит 05

- На экране «Поиск» в поле ввода цвет курсора исправлен на YP_Blue;
- На экране «Медиатека» расстояние между зоной Toolbar'а сделано как на макетах.
- Удалены папки UI через объединение с presentation
